app/upgrade_cudos.go

```diff
@@ -1493,57 +1507,66 @@ func registerBalanceMovement(fromAddress, toAddress string, sourceAmount sdk.Coi
 	manifest.MoveMintedBalance.Movements = append(manifest.MoveMintedBalance.Movements, movement)
 }
 
-func registerManifestMoveDelegations(fromAddress, toAddress string, memo string, manifest *UpgradeManifest) {
+func registerManifestMoveDelegations(fromAddress, toAddress, validatorAddress string, amount sdk.Int, memo string, manifest *UpgradeManifest) {
 	if manifest.MoveDelegations == nil {
 		manifest.MoveDelegations = &UpgradeMoveDelegations{}
 	}
 
 	movement := UpgradeDelegationMovements{
-		From: fromAddress,
-		To:   toAddress,
-		Memo: memo,
+		From:      fromAddress,
+		To:        toAddress,
+		Validator: validatorAddress,
+		Tokens:    amount,
+		Memo:      memo,
 	}
 	manifest.MoveDelegations.Movements = append(manifest.MoveDelegations.Movements, movement)
 	manifest.MoveDelegations.NumberOfMovements = len(manifest.MoveDelegations.Movements)
 }
 
-func moveGenesisDelegations(genesisData *GenesisData, fromAddress, toAddress string, manifest *UpgradeManifest) error {
-	sourceDelegations, exists := genesisData.delegations.Get(fromAddress)
-
-	if !exists {
-		registerManifestMoveDelegations(fromAddress, toAddress, "no_delegations", manifest)
-		// Nothing to move
+func moveGenesisDelegation(genesisData *GenesisData, fromDelegatorAddress, toDelegatorAddress string, validatorAddress string, amount sdk.Int, manifest *UpgradeManifest, memo string) error {
+	// Nothing to move
+	if fromDelegatorAddress == toDelegatorAddress {
 		return nil
 	}
 
-	if destDelegation, destDelegationExists := genesisData.delegations.Get(toAddress); destDelegationExists {
-		// Add delegations - destination delegator has some delegations
-
-		// List all validator addresses in source delegation
-		for _, validatorAddr := range sourceDelegations.Keys() {
-			srcCoins := sourceDelegations.MustGet(validatorAddr)
+	sourceDelegations, exists := genesisData.delegations.Get(fromDelegatorAddress)
+	if !exists {
+		return fmt.Errorf("genesis source delegations of %s not found", fromDelegatorAddress)
+	}
 
-			// Same validator exists in destination delegation - Add coins
-			if destCoins, destValidatorExists := destDelegation.Get(validatorAddr); destValidatorExists {
-				destDelegation.Set(validatorAddr, destCoins.Add(srcCoins...))
+	sourceAmount, exists := sourceDelegations.Get(validatorAddress)
+	if !exists {
+		return fmt.Errorf("genesis source delegation of %s to specific validator %s not found", fromDelegatorAddress, validatorAddress)
```

Comment:
Analogical to [this comment](https://github.com/fetchai/fetchd/pull/387/files#r1793522610).

Though I'm my sensation of valid philosophical bent here is starting to grow.

Reply:
Ignore above, see [here](https://github.com/fetchai/fetchd/pull/387/files/49a2dc388c09dbeda8ce3eb50164d30e7f265787..c869819529f383d7ee7e0c1c382a612f79191269#r1793558935).